### PR TITLE
Code-3051: Use List instead of Vector

### PR DESCRIPTION
**Pull request by grimreaper**
Summary:
This is a portion of my prior pull request designed to be minimal,
reviewable, and correct :/

Test Plan:
Open NameGenPanel in gmgen and look around, try some things.

### CODE REVIEW

File: code/src/java/pcgen/gui2/doomsdaybook/NameGenPanel.java [hunk 1]
```diff
@@ -690,7 +687,7 @@ private void loadCatalogDD()
 			join.retainAll(sexes);
 			Collections.sort(join, new DataElementComperator());
 
-			Vector<RuleSet> catalogs = new Vector<RuleSet>();
+			List<RuleSet> catalogs = new ArrayList<>();
```
Comment (Zaister) [inline code/src/java/pcgen/gui2/doomsdaybook/NameGenPanel.java]: Why is the ArrayList generic here? That should be avoided if possible.

File: code/src/java/pcgen/gui2/doomsdaybook/NameGenPanel.java [hunk 2]
```diff
@@ -724,9 +721,9 @@ private void loadCatalogDD()
 	}
 
 	//	Get a list of all the gender categories in the category map
-	private Vector<String> getGenderCategoryNames()
+	private List<String> getGenderCategoryNames()
 	{
-		Vector<String> genders = new java.util.Vector<String>();
+		List<String> genders = new ArrayList<>();
```
Comment (Zaister) [inline code/src/java/pcgen/gui2/doomsdaybook/NameGenPanel.java]: Same here. Please avoid generics.

File: code/src/java/pcgen/gui2/doomsdaybook/NameGenPanel.java [hunk 3]
```diff
@@ -749,8 +746,8 @@ private void loadCatalogDD()
 	//	Load the gender drop dowd
 	private void loadGenderDD()
 	{
-		Vector<String> genders = getGenderCategoryNames();
-		Vector<String> selectable = new Vector<String>();
+		List<String> genders = getGenderCategoryNames();
+		List<String> selectable = new ArrayList<>();
```
Comment (Zaister) [inline code/src/java/pcgen/gui2/doomsdaybook/NameGenPanel.java]: Again.

File: code/src/java/pcgen/gui2/doomsdaybook/NameGenPanel.java [hunk 4]
```diff
@@ -855,9 +852,9 @@ private void loadData(File path)
 	}
 
 	//	Get a list of category names from the categories map
-	private Vector<String> getCategoryNames()
+	private List<String> getCategoryNames()
 	{
-		Vector<String> cats = new java.util.Vector<String>();
+		List<String> cats = new ArrayList<>();
```
Comment (Zaister) [inline code/src/java/pcgen/gui2/doomsdaybook/NameGenPanel.java]: And here.

File: code/src/java/pcgen/gui2/doomsdaybook/NameGenPanel.java (hunk 1 above)
Comment (grimreaper) [inline code/src/java/pcgen/gui2/doomsdaybook/NameGenPanel.java]: Why should we avoid the diamond operator? It provides more readable, and less verbose code.

Comment (Zaister) [inline code/src/java/pcgen/gui2/doomsdaybook/NameGenPanel.java]: I didn't say avoid it, but if you use ArrayList, say what kind of ArrayList it is. In this case ArrayList<RuleSet>(), i.e. not as a generic.

Comment (Zaister) [inline code/src/java/pcgen/gui2/doomsdaybook/NameGenPanel.java]: On second thought, I see what you are going after, but PCGen's coding style so far has been to spell out the arguments of the generics, and until that style is changed, it should be followed.

**Comment by grimreaper** [comment]
closing until I have a chance to actually fix this.

**Pull request closed without merging**